tests/test_backend.py

```diff
@@ -1,14 +1,29 @@
+from sys import implementation
+from datetime import datetime
+
 import pytest
 
 from openeo_aggregator.backend import AggregatorCollectionCatalog, AggregatorProcessing, \
     AggregatorBackendImplementation, _InternalCollectionMetadata, JobIdMapping
 from openeo_aggregator.caching import DictMemoizer
 from openeo_aggregator.testing import clock_mock
-from openeo_driver.errors import OpenEOApiException, CollectionNotFoundException, JobNotFoundException
+from openeo_driver.backend import ServiceMetadata
+from openeo_driver.errors import OpenEOApiException, CollectionNotFoundException, JobNotFoundException, \
+    ServiceNotFoundException
 from openeo_driver.testing import DictSubSet
 from openeo_driver.users.oidc import OidcProvider
-from .conftest import DEFAULT_MEMOIZER_CONFIG
+from openeo_driver.users.auth import HttpAuthHandler
+from openeo_driver.errors import ProcessGraphMissingException, ProcessGraphInvalidException, ServiceUnsupportedException
+from openeo.capabilities import ComparableVersion
+from openeo.rest import OpenEoApiError, OpenEoRestError
+from .conftest import DEFAULT_MEMOIZER_CONFIG, set_backend_to_api_version
+
 
+TEST_USER = "Mr.Test"
```

Comment:
Good point.

In fact, the corresponding parts of test_views.py and test_backend.py are already testing very similar scenarios from two different points of view, one time directly on the AggregatorSecondaryServices and then again in the views using HTTP requests. Maybe that is useful, but it could just as well be too much duplication.

The main difficulty that I see here to be able to avoid extra auth setup in test_backend.py is that in the AggregatorSecondaryServices we are basically proxying the other services via BackendConnections. We are not directly using other instances of  SecondaryServices, in which case we could simply pass the user_id argument to the other instance. 
As a result I had to use the Flask request to pass on the authentication info to the backend connection.

Maybe there is a better way to do that, but with this method of passing authentication to the backend connection I haven't found a better way to set up authentication in the tests. Essentially it needs a request context so that authentication headers can be passed to the connection.